### DISCUSSION TITLE
Updating the text in MonacoEditorReactComp

### DISCUSSION THREAD

**Discussion by Lemour-sudo** [TypeFox/monaco-languageclient#703] [Q&A]
Is it possible to exit the text in the `MonacoEditorReactComp` through an external method like a state function?
I am trying to enable resetting the the text in the `MonacoEditorReactComp` through a react state but cannot find a way without re-rendering the whole component which makes the editing weird (the cursor resets to the initial position:

```
export function MonacoEditorReactApp() {
  const [code, setCode] = useState(initiaLCode);

  return (
    <div>
      <MonacoEditorReactComp
        userConfig={createLangiumConfig(code)}
        style={{
          paddingTop: "5px",
          height: "80vh",
        }}
        onTextChanged={({ main }) => setCode(main)}
      />
      <button onClick={() => setCode(initiaLCode)}>Reset</button>
    </div>
  );
}
```

**Comment by kaisalmen** [comment]
Hi @Lemour-sudo I think this is related to this https://github.com/TypeFox/monaco-languageclient/issues/697#issuecomment-2227315555 There must be a more easy way to issue and update of the component. The re-evaluation logic for just update or restart of the wrapper/editor is already present, but there is no easy way to trigger it.
